### PR TITLE
feat: add azure storage provider support

### PR DESCRIPTION
**Pull request by Steffen911**
## What

Add support for Azure Blob Store instead of AWS S3.

----

> [!IMPORTANT]
> Adds Azure Blob Storage support as an alternative to AWS S3, with environment configurations, service implementations, and test updates.
> 
>   - **Behavior**:
>     - Adds Azure Blob Storage support as an alternative to AWS S3, controlled by `LANGFUSE_USE_AZURE_BLOB` in `env.ts`.
>     - Updates `StorageServiceFactory` in `StorageService.ts` to return `AzureBlobStorageService` or `S3StorageService` based on environment variable.
>     - Modifies `processEventBatch.ts`, `clickhouse.ts`, `evalService.ts`, `ingestionQueue.ts`, and `legacyIngestionQueue.ts` to use `StorageServiceFactory`.
>   - **Environment**:
>     - Adds `.env.dev-azure.example` for Azure-specific environment variables.
>     - Updates `pipeline.yml` to include Azure Blob Storage in CI/CD tests.
>     - Introduces `docker-compose.dev-azure.yml` for Azure Blob Storage setup.
>   - **Dependencies**:
>     - Adds `@azure/storage-blob` to `package.json` in `shared` package.
>   - **Testing**:
>     - Updates `evalService.test.ts` and `network.ts` to handle Azure Blob Storage scenarios.
>     - Increases test timeouts in `evalService.test.ts` to accommodate Azure Blob Storage operations.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=langfuse%2Flangfuse&utm_source=github&utm_medium=referral)<sup> for 3a3268694001920b157d8b634f03705700274e04. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: packages/shared/src/server/services/StorageService.ts [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+    try {
+      await this.client.createIfNotExists();
+    } catch (err) {
+      logger.error(
+        `Failed to create Azure Blob Storage container ${this.container}`,
+        err,
+      );
+      throw Error("Failed to create Azure Blob Storage container ");
+    }
+  }
```
Comment (Steffen911) [inline packages/shared/src/server/services/StorageService.ts:107]: This is quite convenient during testing and for local development. I'm unsure, whether we want to keep it for production use, though. Essentially this takes whatever bucket name Langfuse is configured with and creates it if missing.

Toggling this behaviour with an environment variable may be an option, but ugly as well.

File: packages/shared/src/server/services/StorageService.ts [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+        err,
+      );
+      throw Error(
+        "Failed to generate presigned upload URL for Azure Blob Storage",
+      );
+    }
+  }
+}
+
+class S3StorageService implements StorageService {
```
Comment (Steffen911) [inline packages/shared/src/server/services/StorageService.ts:279]: Everything from here on and below is taken 1:1 from the old S3StorageService.

File: .env.dev-azure.example [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+LANGFUSE_ENABLE_EXPERIMENTAL_FEATURES="true"
+
+# Salt for API key hashing
+SALT="salt"
+
+# Email
+EMAIL_FROM_ADDRESS=""  # Defines the email address to use as the from address.
+SMTP_CONNECTION_URL=""  # Defines the connection url for smtp server.
+
+# DON'T PANIC: The Azurite Secrets are well-known and meant to be hard-coded
```
Comment (Steffen911) [inline .env.dev-azure.example:37]: Highlighting this part again.

**Review by hassiebp — APPROVED** [review]
All good from media upload side 👍🏾

**Pull request merged**